app/src/pages/Play/YourTurnContent.tsx

```diff
@@ -335,13 +388,18 @@ function YourTurnContent(props: {
                       )
                     ) || null
                   if (nextActiveCard) {
-                    const nextMap = new Map(
-                      shownCardsInActiveTurn.set(
-                        nextActiveCard.id,
-                        ShownCardStatus.Incomplete
-                      )
+                    setShownCardsInActiveTurn(
+                      new Map([
+                        [
+                          nextActiveCard.id,
+                          {
+                            status: ShownCardStatus.Incomplete,
+                            startedAt: new Date(),
+                            endedAt: null
+                          }
+                        ]
+                      ])
```

Comment:
this is the the start of the turn, initialize a brand new map, with 1 incomplete card in it.